Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -13396,15 +13443,30 @@ - (void)writingToolsSession:(WTSession *)session didReceiveAction:(WTAction)acti
     [_webView writingToolsSession:session didReceiveAction:action];
 }
 
-- (void)addTextAnimationForAnimationID:(NSUUID *)uuid withStyleType:(WKTextAnimationType)styleType
+static inline WKTextAnimationType toWKTextAnimationType(WebCore::TextAnimationType style)
 {
+    switch (style) {
+    case WebCore::TextAnimationType::Initial:
+        return WKTextAnimationTypeInitial;
+    case WebCore::TextAnimationType::Source:
+        return WKTextAnimationTypeSource;
+    case WebCore::TextAnimationType::Final:
+        return WKTextAnimationTypeFinal;
+    }
+}
+
+- (void)addTextAnimationForAnimationID:(NSUUID *)uuid withData:(const WebCore::TextAnimationData&)data
+{
+    if (data.style == WebCore::TextAnimationType::Final)
+        [_sourceAnimationIDtoDestinationAnimationID setObject:uuid forKey:data.sourceAnimationUUID];
```

Comment:
This should be moved after the early return due to the preference being disabled.